src/Component/CodeEditor/CodeEditor.tsx

```diff
@@ -343,6 +368,25 @@ export class CodeEditor extends React.Component<CodeEditorProps, CodeEditorState
           >
             {this.getParserOptions()}
           </Select>
+          {activeParser &&
+          activeParser.sldVersion &&
+          activeParser.sldVersion !== '1.0.0' &&
+            <>
+              <span className="symbolizer-units-label">
+                {locale.symbolizerUnitsLabel}:
+              </span>
+              <Select
+                className="gs-code-editor-format-select"
+                style={{ width: 100 }}
+                onSelect={this.onUnitSelect}
+                value={symbolizerUnit}
+              >
+                <Option value="pixel">{locale.symbolizerUnitsPixel}</Option>
+                <Option value="meter">{locale.symbolizerUnitsMeter}</Option>
+                <Option value="foot">{locale.symbolizerUnitsFoot}</Option>
+              </Select>
+            </>
```

Comment:
Please move this to a seperate component.

Reply:
a label and a select? are you sure you want me to do that?

Reply:
Definetly yes. Its a component by definition so it should have its own file. We have many components with less special handling: compare https://github.com/geostyler/geostyler/blob/master/src/Component/Symbolizer/Field/RotateField/RotateField.tsx

Reply:
well looking at the code above and below i dont get the point. there is another similar select, not a standalone component. and there are 2 buttons, no standalone components

Reply:
i refactored the select to a standalone component